### PR TITLE
Fixed possible deadlock when updating route permissions

### PR DESCRIPTION
**Pull request by kozlovic**
This bug is only in master, not in any public release.

Signed-off-by: Ivan Kozlovic <ivan@synadia.com>

/cc @nats-io/core

### CODE REVIEW

File: server/route.go [hunk 1]
```diff
@@ -351,6 +351,16 @@ func (c *client) sendConnect(tlsRequired bool) {
 
 // Process the info message if we are a route.
 func (c *client) processRouteInfo(info *Info) {
+	// We may need to update route permissions and will need the account
+	// sublist. Since getting the account requires server lock, do the
+	// lookup now.
+
+	// FIXME(dlc) - Add account scoping.
+	gacc := c.srv.globalAccount()
```
Comment (derekcollison) [inline server/route.go]: Safe to reference c.srv without capturing it first under a client lock?

Comment (kozlovic) [reply server/route.go]: Yes. This is assigned when client structure is created and never set to nil. Remember, I had argued that we should fix the tests that create client object directly (so possibly without c.srv value) than have the whole server code alway checking for c.srv != nil.

Comment (kozlovic) [reply server/route.go]: which then makes you wonder if it is possible to be nil and does it need to be captured under lock :-)

Comment (derekcollison) [reply server/route.go]: we should probably do that then longer term.

**Comment by derekcollison** [comment]
LGTM

**Pull request merged**